compact.go

```diff
@@ -673,6 +680,50 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 	return nil
 }
 
+func mergeChunks(chks []chunks.Meta) []chunks.Meta {
+	newChks := make([]chunks.Meta, 0, len(chks))
+	for i := 0; i < len(chks); i++ {
+		if i < len(chks)-1 && chks[i].Chunk.NumSamples()+chks[i+1].Chunk.NumSamples() <= 480 {
```

Comment:
This would be easier to read like this:

```go
for i := 0; i < len(chks); i++ {
  if i >= len(chks) - 1 || chks[i].Chunk.NumSamples()+chks[i+1].Chunk.NumSamples() > 480 {
    newChks = append(newChks, chks[i])
    continue
  }
  newChunk := chunkenc.NewXORChunk()
  ...
}
```